pyvptree/include/ISerializable.hpp

```diff
@@ -31,6 +61,10 @@ struct SerializedState {
 
     bool isValid() const { return checksum == calculate_check_sum(data); }
 
+    void buildChecksum() { checksum = calculate_check_sum(data); }
+
+    friend std::ostream &operator<<(std::ostream &os, const SerializedState &state);
+
     private:
     uint8_t calculate_check_sum(const std::vector<uint8_t> &data) const {
         // create an uint8 bitmaks in which bits are alternating 0 and 1
```

Comment:
if we want to include a checksum, a standard CRC-32 could be used instead
not sure how good this one here is

Reply:
This check sum I have built is terrible. The reason is that I did not care much for it so far. But I agree we can user CRC-32. I think I saw one library you worked on for fast CRC but I cannot find now, do you have any suggestions? Thanks!